inductiva/structures/scenarios/plate_linear_elastic/geometry_utils.py

```diff
@@ -0,0 +1,614 @@
+        center_y (float): y-coordinate of the center.
+    """
+
+    def __init__(self, center_x: float, center_y: float) -> None:
+        """Initializes a Hole object."""
+        self.center_x = center_x
+        self.center_y = center_y
+
+    @abstractmethod
+    def perimeter(self):
+        """Abstract method to calculate the perimeter of the hole."""
+        pass
+
+    @abstractmethod
+    def to_dict(self):
+        """Abstract method to convert the hole properties to a dictionary."""
+        pass
+
+    @abstractmethod
+    def to_occ(self):
+        """Abstract method to convert the hole to OpenCASCADE CAD."""
+        pass
+
+    @abstractmethod
+    def get_boundary_ids(self):
+        """Abstract method to get the IDs of the hole boundaries."""
+        pass
+
+    @abstractmethod
+    def get_mesh_metrics(self):
+        """Abstract method to get the metrics for mesh generation in Gmsh."""
+        pass
+
+
+class CircularHole(Hole):
+    """Circular hole.
+
+    Attributes:
+        hole_type (str): Hole type.
+        radius (float): Hole radius.
+    """
+
+    def __init__(self, center_x: float, center_y: float, radius: float) -> None:
+        """Initializes a CircularHole object."""
+        super().__init__(center_x, center_y)
+        self.hole_type = "circular"
+        self.radius = radius
+
+    def perimeter(self) -> float:
+        """Calculate the perimeter of the hole.
+
+        Returns:
+            float: The calculated perimeter.
+        """
+        return 2 * math.pi * self.radius
+
+    def to_dict(self) -> dict:
+        """Convert hole properties to a dictionary.
+
+        Returns:
+            dict: Hole properties.
+        """
+        return {
+            "hole_type": self.hole_type,
+            "center_x": self.center_x,
+            "center_y": self.center_y,
+            "radius": self.radius
+        }
+
+    def to_occ(self) -> int:
+        """Converts hole to OpenCASCADE CAD representation.
+
+        Returns:
+          hole_gmsh (int): The Gmsh entity ID representing the hole's 
+            OpenCASCADE CAD representation.
+        """
+        hole_gmsh = gmsh.model.occ.addDisk(xc=self.center_x,
+                                           yc=self.center_y,
+                                           zc=0,
+                                           rx=self.radius,
+                                           ry=self.radius)
+        return hole_gmsh
+
+    def get_boundary_ids(self, hole_gmsh: int) -> List[int]:
+        """Gets the IDs of the hole boundaries.
+
+        Args:
+            hole_gmsh (int): The Gmsh entity ID representing the hole's 
+              OpenCASCADE CAD representation.
+
+        Returns:
+            List[int]: A list of boundary IDs corresponding to the hole's 
+              boundaries.
+        """
+        hole_boundaries = gmsh.model.getBoundary([(2, hole_gmsh)])
+
+        return [[boundary_id[1]][0] for boundary_id in hole_boundaries]
```

Comment:
The same (about this being better a function in mesh_utils.py) applies here. It even is the same code!